File: OneDrive/Dokumen/Aplikasi Toko Akun Premium/services/whatsapp.py
"""
Generator link wa.me untuk fitur "WhatsApp Auto Order".
Tidak ada payment gateway / checkout — order difinalisasi manual oleh admin
lewat chat WhatsApp sesuai requirement.
"""
from urllib.parse import quote


def format_rupiah(amount: int) -> str:
    return f"Rp{amount:,.0f}".replace(",", ".")


def build_order_message(product_name: str, package_name: str, qty: int, total: int) -> str:
    """Susun pesan otomatis persis sesuai template yang diminta."""
    message = (
        "Halo Admin Rifora Premium,\n\n"
        "Saya ingin membeli:\n\n"
        f"Produk : {product_name}\n"
        f"Paket : {package_name}\n"
        f"Jumlah : {qty}\n"
        f"Total : {format_rupiah(total)}\n\n"
        "Mohon diproses.\n\n"
        "Terima kasih."
    )
    return message


def build_wa_link(phone: str, message: str) -> str:
    """phone harus format internasional tanpa '+' atau '0' di depan (mis. 62...)."""
    clean_phone = phone.strip()
    if clean_phone.startswith("0"):
        clean_phone = "62" + clean_phone[1:]
    return f"https://wa.me/{clean_phone}?text={quote(message)}"

File: lib/whatsapp.ts
/** Normalizes an Indonesian phone number ("0822...", "+62822...", "62822...") to bare 62xxxxxxxxxx for wa.me links. */
export function toWaNumber(raw: string) {
  const digits = raw.replace(/\D/g, "");
  if (digits.startsWith("62")) return digits;
  if (digits.startsWith("0")) return "62" + digits.slice(1);
  return "62" + digits;
}

export function buildOrderMessage(opts: {
  storeName: string;
  brandName: string;
  qty: number;
}) {
  const { storeName, brandName, qty } = opts;
  return [
    `Halo Admin *${storeName}* 👋`,
    ``,
    `Saya ingin order:`,
    `🛍️ Produk: *${brandName}*`,
    `🔢 Jumlah: ${qty}`,
    ``,
    `Mohon info paket, harga, dan langkah pembayarannya ya, terima kasih 🙏`,
  ].join("\n");
}

export function buildCartMessage(opts: {
  storeName: string;
  items: { brandName: string; qty: number }[];
}) {
  const { storeName, items } = opts;
  return [
    `Halo Admin *${storeName}* 👋`,
    ``,
    `Saya ingin order beberapa produk sekaligus:`,
    ``,
    ...items.map((i, idx) => `${idx + 1}. *${i.brandName}* (x${i.qty})`),
    ``,
    `Mohon info paket, harga, dan langkah pembayarannya ya, terima kasih 🙏`,
  ].join("\n");
}

export function buildWaLink(phone: string, message: string) {
  return `https://wa.me/${toWaNumber(phone)}?text=${encodeURIComponent(message)}`;
}
